nic_operator_cni_install.sh

```diff
@@ -0,0 +1,313 @@
+#!/bin/bash -x
+
+export RECLONE=${RECLONE:-true}
+export WORKSPACE=${WORKSPACE:-/tmp/k8s_$$}
+export LOGDIR=$WORKSPACE/logs
+export ARTIFACTS=$WORKSPACE/artifacts
+export TIMEOUT=${TIMEOUT:-300}
+export POLL_INTERVAL=${POLL_INTERVAL:-10}
+
+export NIC_OPERATOR_REPO=${SRIOV_NETWORK_DEVICE_PLUGIN_REPO:-http://l-gerrit.mtl.labs.mlnx:8080/mellanox-network-operator}
+export NIC_OPERATOR_BRANCH=${SRIOV_NETWORK_DEVICE_PLUGIN_BRANCH:-master}
+export NIC_OPERATOR_PR=${SRIOV_NETWORK_DEVICE_PLUGIN_PR-''}
+
+export GOPATH=${WORKSPACE}
+export PATH=/usr/local/go/bin/:$GOPATH/src/k8s.io/kubernetes/third_party/etcd:$PATH
+
+export CNI_BIN_DIR=${CNI_BIN_DIR:-/opt/cni/bin/}
+export CNI_CONF_DIR=${CNI_CONF_DIR:-/etc/cni/net.d/}
+export KUBECONFIG=${KUBECONFIG:-/var/run/kubernetes/admin.kubeconfig}
+
+export KERNEL_VERSION=${KERNEL_VERSION:-5.4.0-29-generic}
+export OS_DISTRO=${OS_DISTRO:-ubuntu}
+export OS_VERSION=${OS_VERSION:-20.04}
+
+# generate random network
+export NETWORK=${NETWORK:-"192.168.0"}
+
+#TODO add autodiscovering
+export SRIOV_INTERFACE=${SRIOV_INTERFACE:-auto_detect}
+export VFS_NUM=${VFS_NUM:-4}
+
+echo "Working in $WORKSPACE"
+mkdir -p $WORKSPACE
+mkdir -p $LOGDIR
+mkdir -p $ARTIFACTS
+
+echo "Get CPU architechture"
+export ARCH="amd"
+if [[ $(uname -a) == *"ppc"* ]]; then
+   export ARCH="ppc"
+fi
+
+function download_and_build {
+    status=0
+    if [ "$RECLONE" != true ] ; then
+        return $status
+    fi
+
+    [ -d /var/lib/cni/sriov ] && rm -rf /var/lib/cni/sriov/*
+
+    echo "Download ${NIC_OPERATOR_REPO}"
+
+    rm -rf $WORKSPACE/mellanox-network-operator
+
+    git clone ${NIC_OPERATOR_REPO} $WORKSPACE/mellanox-network-operator
+    pushd $WORKSPACE/mellanox-network-operator
+    echo "applying the missing patches" 
+    cat > $ARTIFACTS/network_operator.patch << EOF
```

Comment:
i dont like this in code patching approach please remove it if possible or move it to a separate file 
(you have all patches in a dedicated folder, and you apply them here)

lets separate between code and data if this cant be avoided.